indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -31,19 +31,24 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private boolean exclusiveLock;
```

Comment:
Yes making it volatile makes sense...serde is preserving the exclusive lock state, I can make it more explicit by putting `@JsonProperty` annotation on `setExclusiveLock()`. Making it part of constructor can help in testing. Is there any other reason for your preference ? Anyways the tasks have to use LockUpgradeAction to set it to true.